config.go

```diff
@@ -60,37 +66,50 @@ func (c *Config) WithDatabase(database string) *Config {
 	return c
 }
 
-// WithAuth helps to specify the Basic Auth username and password
+// WithAuth helps to specify the Basic Auth username and password.
+// Leave them empty if you are in local environment.
 func (c *Config) WithAuth(username, password string) *Config {
 	c.Username = username
 	c.Password = password
 	return c
 }
 
-func (c *Config) WithKeepalive(interval, timeout time.Duration) *Config {
-	c.keepaliveInterval = interval
-	c.keepaliveTimeout = timeout
+// WithKeepalive helps to set the keepalive option.
+//   - time. After a duration of this time if the client doesn't see any activity it
+//     pings the server to see if the transport is still alive.
+//     If set below 10s, a minimum value of 10s will be used instead.
+//   - timeout. After having pinged for keepalive check, the client waits for a duration
+//     of Timeout and if
+func (c *Config) WithKeepalive(time, timeout time.Duration) *Config {
+	keepalive := options.NewKeepaliveOption(time, timeout).Build()
+	c.options = append(c.options, keepalive)
 	return c
 }
 
-func (c *Config) GetEndpoint() string {
-	return fmt.Sprintf("%s:%d", c.Host, c.Port)
+// TODO(yuanbohan): support more tls options
+func (c *Config) WithInsecure(insecure bool) *Config {
+	opt := options.NewTlsOption(insecure)
+	c.tls = &opt
+	return c
 }
 
-func (c *Config) Options() *Options {
-	if c.keepaliveInterval == 0 && c.keepaliveTimeout == 0 {
-		return nil
-	}
-
-	keepalive := NewKeepaliveOptions()
+// WithDialOption helps to specify the dial option
+// which has not been supported by ingester sdk yet.
+func (c *Config) WithDialOption(opt grpc.DialOption) *Config {
+	c.options = append(c.options, opt)
+	return c
+}
 
-	if c.keepaliveInterval != 0 {
-		keepalive.WithInterval(c.keepaliveInterval)
-	}
+func (c *Config) endpoint() string {
+	return fmt.Sprintf("%s:%d", c.Host, c.Port)
+}
 
-	if c.keepaliveTimeout != 0 {
-		keepalive.WithTimeout(c.keepaliveTimeout)
+func (c *Config) build() []grpc.DialOption {
+	if c.tls == nil {
```

Comment:
yes, it behaves the same as the last version. TlsOption with insecure=true, it will use `grpc.WithTransportCredentials(insecure.NewCredentials())` as one of DialOption